mk/Common.mk

```diff
@@ -313,7 +313,7 @@ RISCV_MARCH      = $(CV_SW_MARCH)
 RISCV_CC         = $(CV_SW_CC)
 RISCV_CFLAGS     = $(CV_SW_CFLAGS)
 
-CFLAGS ?= -Os -g -static -mabi=ilp32 -march=$(RISCV_MARCH) -Wall -pedantic $(RISCV_CFLAGS)
+export CFLAGS ?= -Os -g -static -mabi=ilp32 -march=$(RISCV_MARCH) -Wall -pedantic $(RISCV_CFLAGS)
```

Comment:
This is interesting.  Do we *really* want this to be exported?  If I understand this properly, this **not** will change the shell's definition of `CFLAGS`, but it will define it for all instances of the variable in all included Makefiles.

Was this your intention?

@silabs-hfegran and @pascalgouedo do either of you have any thoughts on this?

FWIW, I tried the change locally and it worked fine.

Reply:
是的这里需要`export`，
Yes, 'export' is required here,
If you do not use the `export`, it will not be able to pass the` CFLAGS `to` Core-V-Verif/CV32E40p/SIM/Core/Core-V-Verif/CV32E40P/SIM/Core/Makefile`.
In `Core-V-Verif/CV32E40P/SIM/Core/Core-V-VERIF/CV32E40p/SIM/Core/Makefile` files` -march` options of `CFLAGS` variables` rv32imc`, but we should use `rv32imc_zicsr` option, otherwise there will be the following errors:
![image](https://user-images.githubusercontent.com/49388283/211701415-b1b539d4-eda9-4f0b-9046-a7131cf57a66.png)

Reply:
Hi,

export CFLAGS is not necessary and not the usual way to pass variables between makefiles.
Here we want to pass RISCV_MARCH to bsp, not CFLAGS.

The pb is that what we have done in uvm tb to resolve this was through a new variable in yaml config file but there is no such configuration feature in core tb.
So the march value should be set through an environment variable in terminal before launching make:
setenv CV_SW_MARCH rv32imc_zicsr or
export CV_SW_MARCH=rv32imc_zicsr

It worked for me when running on test on core env.

So we will not merge this "export CFLAGS " as it impacts uvm tb as well.

Cheers,
Pascal.

Reply:
Okay, thank you for your answer!

Reply:
Thanks @pascalgouedo.   Based on this discussion, I will close this PR.

Thank you again for your questions and contribution @guojiazhuxi.  We do not always accept pull-requests, but we always value them.